ttnn/cpp/ttnn/types.hpp

```diff
@@ -59,76 +59,70 @@ struct CoreGrid {
     }
 };
 
-// Keep track of live buffers and the device addresses they were assigned.
-// When a buffer is created, it is provided a buffer_id using get_buf_id().
-// The address for this buffer is assigned to buffer_id when the buffer is asynchronously allocated.
-// When the buffer destructor is called, or the buffer is asynchronously deallocated, the worker thread
-// will look up the address for buffer_id to free memory on device.
-class buffer_address_map {
+// This buffer class is compatible with multithreaded runtime (which lives in tt_eager)
+// It is derived from the tt_metal::Buffer class, but defines its own asynchronous allocation functions
+class Buffer : public tt::tt_metal::Buffer, public std::enable_shared_from_this<Buffer> {
```

Comment:
Could you rename this Buffer to AsyncBuffer for example.

Reply:
For future proofing, I used the generic `Buffer` name for this class. If we're certain that we won't add an additional `ttnn` level functionality to this class and only use it for async, I'm okay with renaming it.